Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -16,10 +16,27 @@ public class OpenFeatureAPI: GlobalEventPublisher {
     public init() {
     }
 
+    public func setProviderAndWait(provider: FeatureProvider) async {
+        await setProviderAndWait(provider: provider, initialContext: nil)
+    }
+
     public func setProvider(provider: FeatureProvider) {
         self.setProvider(provider: provider, initialContext: nil)
     }
 
+    public func setProviderAndWait(provider: FeatureProvider, initialContext: EvaluationContext?) async {
+        var holder: [AnyCancellable] = []
+        await withCheckedContinuation { continuation in
```

Comment:
Should probably add a timeout here?

Reply:
I'm not sure if this can be considered a function in main api of the `OpenFeatureAPI` or it can be considered as a utility function or an extension function.

Reply:
I made this an extension function of `OpenFeatureAPI`, is that what you suggested @vahidlazio?

Regarding the timeout question, we agree to look into this specific topic in a future PR